rs/bitcoin/ckbtc/minter/src/updates/tests.rs

```diff
@@ -0,0 +1,294 @@
+    use crate::test_fixtures::{
+        ecdsa_public_key, get_uxos_response, ignored_utxo, init_args, init_state, ledger_account,
+        mock::MockCanisterRuntime, quarantined_utxo, KYT_CANISTER_ID, MINTER_CANISTER_ID,
+    };
+    use crate::updates::update_balance;
+    use crate::updates::update_balance::{UpdateBalanceArgs, UtxoStatus};
+    use ic_btc_interface::{GetUtxosResponse, Utxo};
+    use ic_btc_kyt::CheckTransactionResponse;
+    use icrc_ledger_types::icrc1::account::Account;
+
+    #[tokio::test]
+    async fn should_not_add_event_when_reevaluated_utxo_still_ignored() {
+        init_state_with_ecdsa_public_key();
+        let account = ledger_account();
+        let ignored_utxo = ignored_utxo();
+        mutate_state(|s| audit::ignore_utxo(s, ignored_utxo.clone()));
+        let events_before: Vec<_> = storage::events().collect();
+
+        let mut runtime = MockCanisterRuntime::new();
+        mock_get_utxos_for_account(&mut runtime, account, vec![ignored_utxo.clone()]);
+        mock_schedule_now_process_logic(&mut runtime);
+
+        let result = update_balance(
+            UpdateBalanceArgs {
+                owner: Some(account.owner),
+                subaccount: account.subaccount,
+            },
+            &runtime,
+        )
+        .await;
+
+        assert_eq!(result, Ok(vec![UtxoStatus::ValueTooSmall(ignored_utxo)]));
+        assert_has_no_new_events(&events_before);
+    }
+
+    #[tokio::test]
+    async fn should_do_kyt_when_reevaluating_ignored_utxo() {
+        init_state_with_ecdsa_public_key();
+        let account = ledger_account();
+        let ignored_utxo = ignored_utxo();
+        mutate_state(|s| audit::ignore_utxo(s, ignored_utxo.clone()));
+        mutate_state(|s| s.kyt_fee = ignored_utxo.value - 1);
+        let events_before: Vec<_> = storage::events().collect();
+
+        let mut runtime = MockCanisterRuntime::new();
+        mock_get_utxos_for_account(&mut runtime, account, vec![ignored_utxo.clone()]);
+        expect_check_transaction_returning(
+            &mut runtime,
+            ignored_utxo.clone(),
+            CheckTransactionResponse::Failed(vec![]),
+        );
+        mock_schedule_now_process_logic(&mut runtime);
+
+        let result = update_balance(
+            UpdateBalanceArgs {
+                owner: Some(account.owner),
+                subaccount: account.subaccount,
+            },
+            &runtime,
+        )
+        .await;
+
+        assert_eq!(result, Ok(vec![UtxoStatus::Tainted(ignored_utxo.clone())]));
+        assert_has_new_events(
+            &events_before,
+            &[checked_utxo_event(ignored_utxo.clone(), false)],
+        );
+        assert!(!read_state(|s| s.has_ignored_utxo(&ignored_utxo)));
+        assert_eq!(
+            read_state(|s| s.utxo_checked_status(&ignored_utxo).cloned()),
+            Some(UtxoCheckStatus::Tainted)
+        );
+    }
+
+    #[tokio::test]
+    async fn should_mint_reevaluated_ignored_utxo() {
```

Comment:
This should only happen if the evaluated UTXO is not tainted!

Reply:
I have the impressions this is already the case, this is tested L.89

```rust
 expect_check_transaction_returning(
            &mut runtime,
            ignored_utxo.clone(),
            CheckTransactionResponse::Passed,
        );
```